content/blog/kubecon-2019/index.md

```diff
@@ -0,0 +1,271 @@
+---
+title: "KubeCon NA 2019: Are you about to break Prod?"
+date: 2019-11-26
+meta_desc: "In this post, we recap and add more details to our KubeCon talk on setting up ephemeral environments. We will highlight how to acceptance test your application and infrastructure using ephemeral environments."
+meta_image: demo-arch.jpg
+authors: ["erin-krengel"]
+tags: ["testing", "ephemeral environments", "KubeCon", "kubernetes"]
+---
+
+A couple of weeks ago, my colleague Sean and I had the opportunity to present "Are you about to break prod? Acceptance Testing with Ephemeral Environments" at KubeCon NA 2019. In this talk, we covered what is an ephemeral environment, how to create one, and then we walked the audience through a concrete example. Given our limited time, we had to move quickly through a ton of information. Today, I will recap our presentation and add a few more details we weren't able to cover.
+
+<!--more-->
+
+As software engineers, our job is to deliver business value. To do this, we need to be delivering software both quickly and reliably.
+
+So the question we ask you is: are you about to break prod? We will all break production at some point because there are things we miss. As Alexandra points out in her tweet, "failures are part of the cost of building and shipping large systems". Building a robust pipeline, allows us to move quickly in the case of failure and gain confidence around making changes to our infrastructure and applications.
+
+{{< tweet 1198373475049623552 >}}
+
+With this in mind, we will use Pulumi and GitLab to build a pipeline that validates both our application, infrastructure, and deployment process.
+
+## Ephemeral Environments
+
+What is an ephemeral environment? It is a short-lived environment that mimics our production environment. To maintain agility, we typically define the boundaries of this environment to only encompass the first-level dependencies of the particular microservice we are deploying. This means we will not spin up every single microservice or piece of infrastructure that's running in production. Yet we may need to spin up some extra pieces infrastructure to be able to properly test our microservice. For example, we may need to create a subscription to pull from a PubSub topic your microservice writes to. This subscription would allow your acceptance tests to pull from a topic in order to validate an outbound message is published.
```

Comment:
```suggestion
What is an ephemeral environment? It is a short-lived environment that mimics our production environment. To maintain agility, we typically define the boundaries of this environment only to encompass the first-level dependencies of the particular microservice we are deploying. This means we do not spin up every single microservice or piece of infrastructure that's running in production. Yet we may need to spin up some extra pieces infrastructure to be able to test our microservice properly. For example, we may need to create a subscription to pull from a PubSub topic your microservice writes to. This subscription would allow your acceptance tests to pull from a topic to validate an outbound message is published.
```